pkg/auth/service.go

```diff
@@ -1165,21 +1184,26 @@ func CheckPermissions(ctx context.Context, node permissions.Node, username strin
 
 					if stmt.Effect == model.StatementEffectDeny {
 						// this is a "Deny" and it takes precedence
+						permAudit.Denied = append(permAudit.Denied, action)
 						return CheckDeny
+					} else {
+						hasPermission = true
+						allowed = CheckAllow
```

Comment:
```suggestion
					hasPermission = true
					allowed = CheckAllow
```
is closer to the previous version.

We prefer to use an "early return" style: errors return immediately - like l. 1188 does; successes keep going with no "else" and no need to indent.